### DISCUSSION TITLE
[2023-11-27] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by somebody1234** [enso-org/enso#8399] [Weekly Demo Day]
Website2:
- Various bug fixes
- Layout and text changes as requested
- New layout for main blog pages:
![image](https://github.com/enso-org/enso/assets/4046547/18868f20-68ff-42a5-8e3b-28e43bed5960)
![image](https://github.com/enso-org/enso/assets/4046547/8e125f36-8001-4a69-9f90-ed6582b52231)
- Add existing blog posts
![image](https://github.com/enso-org/enso/assets/4046547/bf6d74b8-2779-4085-8c07-cc37e14459d4)
- Flesh out formatting for blog posts:
![image](https://github.com/enso-org/enso/assets/4046547/d8628e17-ddef-453a-895d-044fb04713dc)

GUI2:
- #8375 (As suggested by @JaroslavTulach)
- (WIP, review needs to be addressed) #8389
- (new, waiting on review) #8394
- (ready for review, requires testing, no animations due to a bug in Vue) #8361

<img width="287" alt="image" src="https://github.com/enso-org/enso/assets/919491/2fad3391-08e4-48c7-a038-b421bf8b7566">

**Comment by farmaazon** [comment]
* #8316 Finally landed
* #8365 - The prod build of GUI2 app has JS errors, an error in some tool. Applied a workaround.
* #8377
  * Fixed sorting
  * searching by initials now does not skip words
  * icons: now we use same logic as existing nodes.
  * local entities has worse score if "show locals" option is toggled off.
* Also a second set of CB fixes is on its way: #8385 
  * The "main view" (when no self type/module nor pattern is specified) contains all methods defined in top modules of every library
  * Entries defined inside any "Internal" module from stdlib will be hidden by default; they will be visible only when self type is specified or the module path leading to such module internals.
  * Fixed an issue with missing groups sometimes.
  * Modules and local variables have different default icon.
  * Updated the set of icons available in GUI2 to match the icons in the newest design.

![image](https://github.com/enso-org/enso/assets/3919101/cb33691e-222b-413e-a92e-2cf84e9fe4bb)

![image](https://github.com/enso-org/enso/assets/3919101/beab202d-4feb-4b00-ba0c-c141862da53c)

**Comment by kazcw** [comment]
Finished integrating new AST with main app logic (#8263). Fixing some merge conflicts and then I'll work on edits.

**Comment by MichaelMauderer** [comment]
* Worked on  #8259

[Peek 2023-11-27 14-44.webm](https://github.com/enso-org/enso/assets/1428930/49a25fe2-33bf-4c08-9dec-d5fe5b84343a)

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/8400, which turned out to be a needlessly complicated task because of TS`s quirks.
- https://github.com/enso-org/enso/pull/8315 imports were reviewed and merged.

<img width="846" alt="Screenshot 2023-11-27 at 5 24 57 PM" src="https://github.com/enso-org/enso/assets/6566674/c8b4730a-69eb-4efe-8583-8a6dd887d239">

**Comment by mwu-tow** [comment]
* Fixing macOS notarization. https://github.com/enso-org/enso/pull/8392
* Minor fixes / improvements https://github.com/enso-org/enso/pull/8285
  * Support PowerShell 7.4.0 
  * Follow up fixes for Graal bump.

**Comment by Frizi** [comment]
- Fixed regressions related to edge dragging and connecting to ports, added support for multiple output ports from a single node (although full support will need engine to also support that syntax)
- Expanded function call argument recognition to properly handle `self` argument and partially applied functions from a variable #8388 
- Still working on handling connecting edges to placeholders #8258

<img width="865" alt="image" src="https://github.com/enso-org/enso/assets/919491/73126593-05c0-4553-ba6d-dad97d083c48">
<img width="961" alt="image" src="https://github.com/enso-org/enso/assets/919491/ffe02d79-841c-411d-a218-de89c2522f7b">

**Comment by hubertp** [comment]
Relevant GUI changes:
- added shortcuts for starting/stopping profiling #8358 
- Further improvements to startup #8359
